spring-boot-actuator/src/main/java/org/springframework/boot/actuate/info/InfoProvider.java

```diff
@@ -0,0 +1,17 @@
+package org.springframework.boot.actuate.info;
+
+import java.util.Map;
+
+/**
+ * information provider for the info endpoint
+ *
+ * @author Meang Akira Tanaka
+ */
+public interface InfoProvider {
+
+	/**
+	 * @return a collection of information
+	 */
+	Map<String, Object> provide();
```

Comment:
Hi Stephane

I agree that it would be better to return it's own type rather than a basic
type as map.

however i returned the map because the old implementation of the invoke
method at InfoeEndpoint returns a map. So my question was asked if it would
be OK to break the signature of the original method?

> I am not too sure what to do at this point but it feels to me that the
default implementation of InfoProvidershould read the info.* keys from the
context.

I think the infoendpoint is already doing that. Should that be moved to a
default implementation?

Or is the name infoprovider correct?

Br

Meang



On Wed, Jul 15, 2015, 10:06 Stéphane Nicoll <notifications@github.com>
wrote:

> In
> spring-boot-actuator/src/main/java/org/springframework/boot/actuate/info/InfoProvider.java
> <https://github.com/spring-projects/spring-boot/pull/3492#discussion_r34654867>
> :
>
> > @@ -0,0 +1,17 @@
> > +package org.springframework.boot.actuate.info;
> > +
> > +import java.util.Map;
> > +
> > +/**
> > + * information provider for the info endpoint
> > + *
> > + * @author Meang Akira Tanaka
> > + */
> > +public interface InfoProvider {
> > +
> > +	/**
> > +	 * @return a collection of information
> > +	 */
> > +	Map<String, Object> provide();
>
> Break what interface? You just created it :)
>
> If you return a strongly-typed object, it gives you the ability to evolve
> said object in the future without breaking anything. In general returning
> simple types like map from an interface is a bad idea.
>
> I am not too sure what to do at this point but it feels to me that the
> default implementation of InfoProvider should read the info.* keys from
> the context.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spring-projects/spring-boot/pull/3492/files#r34654867>
> .
>